CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+Contributing
+============
+
+If you're here, you would like to contribute to this project and you're really welcome!
+
+Bug reports
+-----------
+
+If you find a bug or a documentation issue, please report it or even better: fix it :). If you report it,
+please be as precise as possible. Please use the provided template when creating an issue. Your issue is 
+more likley to get fast feedback if you are clear and detailed in your report. disclosing the vulnerability.
+
+Feature requests
+----------------
+
+If you think a feature is missing, please report it or even better: implement it :). If you report it, describe the more
+precisely what you would like to see implemented and we will discuss what is the best approach for it. If you can do
+some research before submitting it and link the resources to your description, you're awesome! It will allow us to more
+easily understood/implement it.
+
+Sending a Pull Request
+----------------------
+
+If you're here, you are going to fix a bug or implement a feature and you're the best!
+To do it, first fork the repository, clone it and create a new branch with the following commands:
+
+```bash
+$ git clone git@github.com:your-name/repo-name.git
+$ git checkout -b feature-or-bug-fix-description
+```
+
+Then install the dependencies through Composer:
+
+```bash
+$ composer install
+```
+
+Write code and tests. When you are ready, run the tests.
+
+```bash
+$ phpunit
+```
+
+When you are ready with the code, tested it and documented it, you can commit and push it with the following commands:
+
+```bash
+$ git commit -m "Feature or bug fix description"
+$ git push origin feature-or-bug-fix-description
+```
+
+> Please write your commit messages in the imperative and follow the 
+[guidelines](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) for clear and concise messages.
+
+Then [create a pull request](https://help.github.com/articles/creating-a-pull-request/) on GitHub.
+
+Please make sure that each individual commit in your pull request is meaningful.
+If you had to make multiple intermediate commits while developing,
+please squash them before submitting with the following commands
+(here, we assume you would like to squash 3 commits in a single one):
+
+```bash
+$ git rebase -i HEAD~3
+```
+
+If your branch conflicts with the master branch, you will need to rebase and re-push it with the following commands:
+
+```bash
+$ git remote add upstream git@github.com:orga/repo-name.git
+$ git pull --rebase upstream master
+$ git push -f origin feature-or-bug-fix-description
+```
+
```

Comment:
You do never need to amend commits or rewrite the history anymore. Since the 1st of April we can squash the commits when merging.